lib/src/cpp/utils/ParseXML.cpp

```diff
@@ -399,18 +431,37 @@ std::string ParseXML::get_absolute_path(
     return absolute_xml_file;
 }
 
+std::vector<uint>* ParseXML::get_real_index(
+        xercesc::DOMNodeList*& node_list)
+{
+    // Create new list
+    std::vector<uint>* index_list = new std::vector<uint>();
+
+    // Iterate through the list
+    for (uint i = 0, size = node_list->getLength(); i < size; i++)
+    {
+        if (node_list->item(i)->getNodeType() == xercesc::DOMNode::NodeType::ELEMENT_NODE)
+        {
+            index_list->push_back(i);
+        }
+    }
+
+    // return list
+    return index_list;
+}
+
 xercesc::DOMNode* ParseXML::get_node(
         xercesc::DOMDocument*& doc,
         const std::string& tag_name)
 {
     // Try call main get_node function with remain empty values
-    return get_node(doc, tag_name, 0, "", "");
+    return get_node(doc, tag_name, nullptr, "", "");
```

Comment:
```suggestion
    return get_node(doc, tag_name, "", "", "");
```

If accepted the suggestion of passing the index as reference and not as pointer.

Reply:
This would be eventually modified